paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/BaseDataTableSource.java

```diff
@@ -182,6 +200,29 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
                                 .build());
     }
 
+    private ScanRuntimeProvider createCountStarScan() {
+        TableScan scan = table.newReadBuilder().withFilter(predicate).newScan();
+        List<PartitionEntry> partitionEntries = ((DataTableBatchScan) scan).planPartitions();
+        long rowCount = partitionEntries.stream().mapToLong(PartitionEntry::recordCount).sum();
+        NumberSequenceRowSource source = new NumberSequenceRowSource(rowCount, rowCount);
```

Comment:
inline new NumberSequenceRowSource(rowCount, rowCount); to SourceProvider#createSource()?

Reply:
No need.